Chiu Chi Kuan ip

zlimez opened the pull request:
# Add **CRUD** functionalities 
Zlimez, a task management app, helps you track the countless number of tasks you have to do while encouraging you to still make good use of your memory so that you do not become a goldfish someday. 
> "You can’t trust other people. If it’s important, you have to do it yourself" - _Neil Gaiman_

Download from [here](https://github.com/zlimez/ip/releases/tag/A-Jar) to play around and see for yourself the temperament of Zlimez and what he can do for you 😡.

Current features include:
- Add tasks
- Mark tasks
- Delete tasks
- List tasks _(All or those on a specific date)_
- Find tasks

Event types currently supported include:
1. General todo tasks
2. Events
3. Tasks with deadline

Long term product goals:
- [x] Task Management
- [ ] GUI (coming soon)
- [ ] Conversational

A snapshot of Zlimez's response, assuming you understand Java:
```
  public void bye() {
      System.out.println("\tBye. zzz FINALLY~~" + " " + EMOJI);
      reader.close();
  }
```

Commit:
Add Gradle support

zlimez committed:
=Greet and Echo

zlimez committed:
=Add and List

zlimez committed:
=Mark and Unmark

zlimez committed:
=Different Tasks

zlimez committed:
=Error Handling

zlimez committed:
=Semi-automated Testing

zlimez committed:
=Delete

zlimez committed:
=Action enum

zlimez committed:
=Action enum

zlimez committed:
=Conflict resolving

zlimez committed:
=Save and init data

zlimez committed:
=Date Treatment

zlimez committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/chatbot/main/Duke.java
#	src/main/java/chatbot/tasks/Deadline.java
#	src/main/java/chatbot/tasks/Event.java

zlimez committed:
=Restructure to fit OOP paradigm

zlimez committed:
=Place classes into packages

zlimez committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

zlimez committed:
=Add test for parser and tasks

zlimez committed:
=Change build.gradle

zlimez committed:
=Add JavaDoc

zlimez committed:
=Add find function

zlimez committed:
=Tweak to fit coding standards

zlimez committed:
Merge branch 'branch-Level-9'

zlimez committed:
=Resolve conflict master and A-JavaDoc

zlimez committed:
=Add JavaDoc to find related functions and class

highorbit25 commented on src/main/java/chatbot/storage/Storage.java:
```diff
@@ -0,0 +1,86 @@
+ * before it ends.
+ */
+public class Storage {
+    private static final String DIVIDER = " \\| ";
+    private static final String ROOT = System.getProperty("user.dir");
+    private static final Path DATA_PATH = Paths.get(ROOT, "data", "data.txt");
```
Good job, defining these constants improved code readability greatly.

zlimez replied:
Thanks

highorbit25 commented on text-ui-test/input.txt:
```diff
@@ -0,0 +1,19 @@
+mark 2
+mark 1
+unmark 1
+deadline kiss you /at 2022-09-15
+deadline kiss you /by 2022-09-15
```
HAHA 💯

highorbit25 commented on src/main/java/chatbot/tasks/TaskList.java:
```diff
@@ -0,0 +1,164 @@
+import chatbot.exceptions.DukeException;
+
+import java.time.LocalDate;
+import java.util.*;
```
Perhaps can try to avoid wildcard imports

zlimez replied:
Yap haha just included checkstyle and found that out

highorbit25 commented on src/main/java/chatbot/ui/UI.java:
```diff
@@ -0,0 +1,137 @@
+        System.out.println("\tBye. zzz FINALLY~~" + " " + EMOJI);
+        reader.close();
+    }
+}
```
Generally well-written code with no major code quality issues! Keep up the good work!

zlimez committed:
=Add Checkstyle plugin

zlimez committed:
=Add GUI Skeleton

zlimez committed:
=Polish GUI related logic

zlimez committed:
=Merge GUI into master

richavm14 commented on src/main/java/chatbot/parser/Parser.java:
```diff
@@ -0,0 +1,196 @@
+     * @throws DukeException Raised if the argument does not contain
+     *      a task name and a date.
+     */
+    public static String[] parseTimedTask(String type, String args) throws DukeException {
```
I like the use of Egyptian style braces and whitespace

richavm14 commented on src/main/java/chatbot/parser/Parser.java:
```diff
@@ -0,0 +1,196 @@
+ * The class is responsible for parsing user input and determining the consequent action.
+ */
+public class Parser {
+    public static final Pattern EVENT_FORMAT = Pattern.compile("(?<taskName>.+)\\s+/at\\s+(?<date>\\S+)");
```
I like that you followed naming convention

richavm14 commented on src/main/java/chatbot/commands/Unmark.java:
```diff
@@ -0,0 +1,44 @@
+import chatbot.tasks.Task;
+import chatbot.tasks.TaskList;
+import chatbot.ui.Response;
+import chatbot.ui.UI;
```
I like that you kept the order of imports consistent!

richavm14 commented on src/main/java/chatbot/tasks/Task.java:
```diff
@@ -0,0 +1,48 @@
+ * associated.
+ */
+public abstract class Task {
+    private boolean isComplete = false;
```
Boolean variable is named to sound like boolean 👍

richavm14 reviewed (commented):
You have followed coding standards. Good work!

zlimez committed:
=Add Assertions

zlimez committed:
=Improve code quality

zlimez committed:
=Add stream variant for find

zlimez committed:
Merge branch 'branch-A-Assertions' into branch-A-Streams

zlimez committed:
Merge pull request #2 from zlimez/branch-A-Assertions

=Add Assertions

zlimez committed:
Merge branch 'branch-A-CodeQuality'

zlimez committed:
Merge branch 'branch-A-Streams'

zlimez committed:
=Add tagging

Users could be inclined to attach tags to their tasks, e.g. #important #work.

Let's add a new feature for the chat bot, tagging. User can attach multiple
tags to a task via appending the tags after the a task creation command,
e.g. todo CS2103 Quiz #schoolwork #online. Consequently, users can also query
for tasks that are given a specified tag e.g. findtag important.

This enables finer classification of tasks.

zlimez committed:
=Add tagging

zlimez committed:
=Polish Tag related functions

zlimez committed:
Setup CI

Let's include a .yml file such that every push and pull action will cause the automated
JUnit tests to run and help developers ensure program correctness.

zlimez committed:
=Improve UI and add list tags feature

zlimez committed:
=Enhance personality and minor fixes

zlimez committed:
=Include Ui snapshot

zlimez committed:
=Include Ui snapshot

zlimez committed:
Merge branch 'master' of https://github.com/zlimez/ip

zlimez committed:
Update README.md (Incomplete)

zlimez committed:
=Add UG related resources

zlimez committed:
Merge branch 'master' of https://github.com/zlimez/ip

zlimez committed:
=Complete User guide

zlimez committed:
=Rename jar output
